hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBCheckpointServlet.java

```diff
@@ -79,14 +86,27 @@ public void initialize(DBStore store, DBCheckpointMetrics metrics,
 
     this.aclEnabled = omAclEnabled;
     this.ozAdmins = ozoneAdmins;
+    this.isSpnegoEnabled = isSpnegoEnabled;
+    ReconConfig reconConfig = conf.getObject(ReconConfig.class);
+    String reconPrincipal = reconConfig.getKerberosPrincipal();
+    if (!reconPrincipal.isEmpty()) {
+      UserGroupInformation ugi =
+          UserGroupInformation.createRemoteUser(reconPrincipal);
+      reconSPN = ugi.getShortUserName();
+    } else {
+      reconSPN = StringUtils.EMPTY;
```

Comment:
Do we have guarantee that SCM configuration always have recon related configuration like recon SPN?